apps/tx-builder/src/components/forms/fields/JsonField.tsx

```diff
@@ -0,0 +1,171 @@
+import { useState, useCallback, ClipboardEvent } from 'react';
+import styled from 'styled-components';
+import { Icon, TextFieldInput, Tooltip, GenericModal, Text, Button, IconTypes } from '@gnosis.pm/safe-react-components';
+import IconButton from '@material-ui/core/IconButton';
+import { Box } from '@material-ui/core';
+
+const DEFAULT_ROWS = 4;
+
+type Props = {
+  id: string;
+  name: string;
+  label: string;
+  value: string;
+  onChange: (value: string) => void;
+};
+
+const JsonField = ({ id, name, label, value, onChange }: Props) => {
+  const [showReplaceModal, setShowReplaceModal] = useState(false);
+  const [tempAbi, setTempAbi] = useState(value);
+  const [isPrettified, setIsPrettified] = useState(false);
+
+  const toggleFormatJSON = useCallback(() => {
+    if (!value) {
+      return;
+    }
+
+    try {
+      onChange?.(JSON.stringify(JSON.parse(value), null, isPrettified ? 0 : 2));
+      setIsPrettified(!isPrettified);
+    } catch (e) {
+      console.error(e);
+      onChange(value);
+    }
+  }, [onChange, value, isPrettified]);
+
+  const toggleModal = useCallback(() => setShowReplaceModal(!showReplaceModal), [showReplaceModal]);
+
+  const changeAbi = useCallback(() => {
+    onChange?.(tempAbi);
+    setIsPrettified(false);
+    toggleModal();
+  }, [tempAbi, onChange, toggleModal]);
+
+  const handlePaste = useCallback(
+    (event: ClipboardEvent<HTMLInputElement>) => {
+      event.preventDefault();
+      event.stopPropagation();
+
+      const clipboardData = event.clipboardData;
+      const pastedData = clipboardData?.getData('Text') || '';
+
+      if (value && pastedData) {
+        setTempAbi(pastedData);
+        toggleModal();
+      } else {
+        onChange(pastedData);
+      }
+    },
+    [onChange, toggleModal, value],
+  );
+
+  return (
+    <>
```

Comment:
is `<>` really needed here?

Reply:
yep because I have a Modal under the JSON Field